### DISCUSSION TITLE
Tasklist icons not displayed

### DISCUSSION THREAD

**Discussion by qumpa** [awesomeWM/awesome#3875] [General]
I made myself a custom icon theme and it works perfectly in other desktop environments but in awesomewm few icons are missing altogether and few others are what appears to be default icon (gnome, hicolor or some other theme that is shipped with distro)

Missing icons are: feh, freetube
Default ones: sublime, smplayer

There are probably more but those are the ones I spotted so far.

Oddly enough all icons are displayed correctly in Gnome System Monitor which leads me to believe it must have something to do with awesome.

**Comment by sclu1034** [comment]
Awesome uses the icon that the client assigns via X properties. Any kind of universal theming beyond that is your own to implement, even if that's just about querying the current GTK/Qt theme.

**Comment by actionless** [comment]
you could create your own template for tasklist and override the clienticon logic: https://awesomewm.org/apidoc/widgets/awful.widget.tasklist.html

and here example of icon lookup for the current theme (with implementing fallback logic for icons which are missing): https://github.com/actionless/awesome_config/blob/master/actionless/util/xdg.lua#L90
